New dependent-packages std.build feature: plainest-possible 3-project setup (3 libs + 1 exe) won't build, but `--pkg-begin/end`s look fine

I was super glad when #4343 by @frmdstryr landed, and the full command (further below) that `build.zig` stitches together looks right to me, but building fails `unable to find '<dep>'` with a **most-simplistic multi-lib+one-prog setup of 3 side-by-side projects** as in [this maximally-minimized-repro repo](https://github.com/meta-leap/zig-pkg-deps-mini-repro), which you can pull and play with easily: `cd ./proj3` and `zig build` to provoke the issue described.

You see 3 dirs: proj1, proj2, proj3.
- 2nd depends on the 1st,
- 3rd depends on **both** the 1st and 2nd (in the real world calling things from both).

What works & what fails:

- In proj1, `zig build demo` runs, no prob. No deps either.
- In proj2, `zig build demo` runs too. No big dep trees, just a single `.addPackagePath("proj1", "../proj1/api.zig")` suffices for this only-1-dependency project.
- But proj3's [build.zig](https://github.com/meta-leap/zig-pkg-deps-mini-repro/blob/master/proj3/build.zig) now needs to set things up as instructed in #4343: using `.addPackage` and `Pkg`s with `.dependencies` slices set.
  - (Ignore for now here that there are 3x `.addPackage` calls when just the last one alone would suffice and bring in its deps quite nicely --- doesn't change the problem at hand, as follows.)

If you ditch the last 4 lines in `fn build`, ie. the `.addExecutable` block, then `zig build demo` works & runs. But finally if you keep it in as in the linked source, build fails with: 

```
/home/_/c/z/zig-pkg-deps-mini-repro/proj2/src/lib2.zig:1:14: error: unable to find 'proj1'
const lib1 = @import("proj1");
             ^
/home/_/c/z/zig-pkg-deps-mini-repro/proj2/src/lib2.zig:4:12: note: referenced here
    return lib1.isFoo1(T) and isFoo1(T);
           ^
prog3...The following command exited with error code 1:
```

Then follows, here word-wrapped for readability: `/home/_/a/zig/zig build-exe /home/_/c/z/zig-pkg-deps-mini-repro/proj3/proggie/main.zig --cache-dir /home/_/c/z/zig-pkg-deps-mini-repro/proj3/zig-cache --name prog3 --pkg-begin proj3 /home/_/c/z/zig-pkg-deps-mini-repro/proj3/api.zig --pkg-begin proj1 /home/_/c/z/zig-pkg-deps-mini-repro/proj1/api.zig --pkg-end --pkg-begin proj2 /home/_/c/z/zig-pkg-deps-mini-repro/proj2/api.zig --pkg-end --pkg-end --pkg-begin proj2 /home/_/c/z/zig-pkg-deps-mini-repro/proj2/api.zig --pkg-begin proj1 /home/_/c/z/zig-pkg-deps-mini-repro/proj1/api.zig --pkg-end --pkg-end --pkg-begin proj1 /home/_/c/z/zig-pkg-deps-mini-repro/proj1/api.zig --pkg-end --cache on `

What troubles me, the full `build-exe` command (see above) **looks fine to me as far as `--pkg-begin` and `--pkg-end` nestings** are concerned.

If you take the exe out of `proj3/build.zig`, and put a separate `build.zig` under `proj3/proggie` instead, just for the exe and adding the 3 deps? Exact same issue.

What could it be?

Using current nightly binary `0.5.0+0fdcd5c4c` (but the earlier one 0.5.0+5504f155a had this issue too, that's why I switched to current), Linux x86_64.

--- a/lib/std/build.zig
+++ b/lib/std/build.zig
@@ -1775,7 +1775,7 @@ pub const LibExeObjStep = struct {
         }
     }
 
-    fn makePackageCmd(self: *LibExeObjStep, pkg: Pkg, zig_args: *ArrayList([]const u8)) anyerror!void {
+    fn makePackageCmd(self: *LibExeObjStep, pkg: Pkg, zig_args: *ArrayList([]const u8)) error{OutOfMemory}!void {
         const builder = self.builder;
 
         try zig_args.append("--pkg-begin");
@@ -2053,7 +2053,7 @@ pub const LibExeObjStep = struct {
                 try zig_args.append("--test-cmd-bin");
             },
         }
-        
+
         for (self.packages.span()) |pkg| {
             try self.makePackageCmd(pkg, &zig_args);
         }

--- a/lib/std/build.zig
+++ b/lib/std/build.zig
@@ -1775,6 +1775,22 @@ pub const LibExeObjStep = struct {
         }
     }
 
+    fn makePackageCmd(self: *LibExeObjStep, pkg: Pkg, zig_args: *ArrayList([]const u8)) anyerror!void {
+        const builder = self.builder;
+
+        try zig_args.append("--pkg-begin");
+        try zig_args.append(pkg.name);
+        try zig_args.append(builder.pathFromRoot(pkg.path));
+
+        if (pkg.dependencies) |dependencies| {
+            for (dependencies) |sub_pkg| {
+                try self.makePackageCmd(sub_pkg, zig_args);
+            }
+        }
+
+        try zig_args.append("--pkg-end");
+    }
+
     fn make(step: *Step) !void {
         const self = @fieldParentPtr(LibExeObjStep, "step", step);
         const builder = self.builder;
@@ -2037,21 +2053,9 @@ pub const LibExeObjStep = struct {
                 try zig_args.append("--test-cmd-bin");
             },
         }
+        
         for (self.packages.span()) |pkg| {
-            try zig_args.append("--pkg-begin");
-            try zig_args.append(pkg.name);
-            try zig_args.append(builder.pathFromRoot(pkg.path));
-
-            if (pkg.dependencies) |dependencies| {
-                for (dependencies) |sub_pkg| {
-                    try zig_args.append("--pkg-begin");
-                    try zig_args.append(sub_pkg.name);
-                    try zig_args.append(builder.pathFromRoot(sub_pkg.path));
-                    try zig_args.append("--pkg-end");
-                }
-            }
-
-            try zig_args.append("--pkg-end");
+            try self.makePackageCmd(pkg, &zig_args);
         }
 
         for (self.include_dirs.span()) |include_dir| {

